woocommerce/woocommerce
# Product Editor: Add "Copy all content" menu item to description editor modal

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds a "Copy all content" menu item to the description editor modal.

Known issue:
- The transient notice (snackbar) saying that the copy succeeded is (mostly) hidden under the modal. This will be addressed separately (see #39208)

Closes #39183.

<img width="1258" alt="Screenshot 2023-07-12 at 13 41 14" src="https://github.com/woocommerce/woocommerce/assets/2098816/55c9c9c1-8698-46e6-b5b6-efcf43a814a0">




### How to test the changes in this Pull Request:



1. Enable the new block-based product editor.
2. Create a new simple product.
3. Click on "Edit description" to bring up the description modal editor.
4. Enter in a description. Try using some different blocks.
5. Click on the "..." more menu dropdown.
6. Click on the "Copy all content" menu item.
7. Verify that the Tracks event `wcadmin_product_iframe_editor_copy_all_content_menu_item_click ` is logged.
8. Paste in a variety of places and verify the result is correct:
    - Back into description editor... blocks pasted should look identical
    - Into a post editor... blocks pasted should look identical
    - Into a text editor... text should be the standard Gutenberg serialized representation of blocks (HTML with comments)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
